### PR TITLE
Build component with PostCSS

### PR DESCRIPTION
**Pull request by simonsmith**
So this is a stab at building a component with PostCSS in response to - https://github.com/suitcss/preprocessor/issues/11. It emits warnings to the console if the conformance fails, but we can also configure this to throw an error if needs be.

screen shot 2015-10-02 at 23 52 52

~~Hopefully the `postcss-config.js` could be dropped into each component and just work.~~
Each component can use just the plugins it requires.

cc @timkelty @giuseppeg

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,14 +8,22 @@
     "lib"
   ],
   "devDependencies": {
-    "suitcss-components-test": "*",
-    "suitcss-preprocessor": "~0.3.0"
+    "autoprefixer": "^6.0.3",
+    "postcss": "^5.0.8",
+    "postcss-bem-linter": "^1.1.0",
+    "postcss-calc": "^5.0.0",
```
Comment (giuseppeg) [inline package.json]: `Button` is not using `postcss-calc`

File: package.json [hunk 2]
```diff
@@ -8,14 +8,22 @@
...
+    "postcss-cli": "^2.1.0",
+    "postcss-custom-media": "^5.0.0",
```
Comment (giuseppeg) [inline package.json]: Nor `postcss-custom-media`

File: package.json (hunk 1 above)
Comment (simonsmith) [inline package.json]: Good point, we should really only use the plugins that each package uses.

File: package.json [hunk 3]
```diff
@@ -8,14 +8,22 @@
...
+    "postcss-custom-properties": "^5.0.0",
+    "postcss-import": "^7.0.0",
```
Comment (giuseppeg) [inline package.json]: and `import`

Comment (simonsmith) [inline package.json]: We're using this one, as it's imported into `index.css`. The tests make use of it too.

File: package.json (hunk 2 above)
Comment (simonsmith) [inline package.json]: :+1:

File: lib/button.css [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/** @define Button; use strict */
+/** @define Button */
```
Comment (giuseppeg) [inline lib/button.css]: Does `postcss-bem-linter` always run in "strict mode"?

File: package.json (hunk 3 above)
Comment (giuseppeg) [inline package.json]: yep it was too late when I realized that :)

**Comment by giuseppeg** [comment]
I like this approach!
I wonder if it can affect the overall build time of an application with hundreds of components though. Maybe we should benchmark this.

File: lib/button.css (hunk 4 above)
Comment (simonsmith) [inline lib/button.css]: Yeah, you have to opt out of it with 'weak'

**Comment by simonsmith** [comment]
I've built tens rather than hundreds before and it always seems to be fast. PostCSS claims to be one of the faster processors. 

One thing I like here is that each package will compile with just the plugins it needs but it makes very easy for people to add more plugins should they need them.

**Comment by necolas** [comment]
I think moving all the packages into a monorepo should be considered while doing this. Individual components can still be published to their separate npm packages. But having to maintain the toolchain across lots of repos is going to be a hassle for you.

**Comment by simonsmith** [comment]
I agree, it would make this easier if we were happy to give up building of individual components.

**Comment by simonsmith** [comment]
Closing this in favour of the mono-repo approach in progress at suitcss/suit/issues/109

**Pull request closed without merging**